modules/lib/lib-auth/src/main/resources/lib/xp/auth.ts

```diff
@@ -0,0 +1,832 @@
+/**
+ * Built-in authentication functions.
+ *
+ * @example
+ * var authLib = require('/lib/xp/auth');
+ *
+ * @module auth
+ */
+
+declare global {
+    interface XpLibraries {
+        '/lib/xp/auth': typeof import('./auth');
+    }
+}
+
+function required<T extends Object>(obj: T, name: keyof T): void {
+    if (obj == null || obj[name] == null) {
+        throw `Parameter \'${String(name)}\' is required`;
+    }
+}
+
+export type LoginScopeType = 'SESSION' | 'REQUEST' | 'NONE';
+
+export interface LoginParams {
+    user: string;
+    skipAuth?: boolean;
+    password?: string;
+    idProvider?: string;
+    scope?: LoginScopeType;
+    sessionTimeout?: number;
+}
+
+export interface Principal {
+    type: string;
+    key: string;
+    displayName: string;
+    modifiedTime: string;
+}
+
+export interface User
+    extends Principal {
+    disabled?: boolean;
+    email?: string;
+    login?: string;
+    idProvider: string;
+    profile?: Record<string, Object>;
```

Comment:
Just replace it with `object` type, because it is pretty much similar to the current type. `Record` should be used, when we limit available list of keys or values type, e.g. `Record<'foo' | 'bar', number>`

Reply:
Replaced by `Record<string, unknow>`